modules/civicrm_engage/civicrm_engage.module

```diff
@@ -136,7 +136,7 @@ function civicrm_engage_civicrm_pageRun(&$page) {
     }
     $contactId = $page->getVar('_contactId');
     if (empty($contactId)) {
-      $contactId = CRM_Utils_Request::retrieve('cid', 'Positive', $this, TRUE, NULL, $_REQUEST);
+      $contactId = CRM_Utils_Request::retrieve('cid', 'Positive', $contactId, TRUE, NULL, $_REQUEST);
```

Comment:
Looking at the code here https://github.com/civicrm/civicrm-core/blob/master/CRM/Utils/Request.php#L74 i think the more correct option might be to use $page i think

Reply:
you're absolutely right!  I'll change that